test/integration/vaultauthmethods_integration_test.go

```diff
@@ -0,0 +1,223 @@
+				Kubernetes: &secretsv1alpha1.VaultAuthConfigKubernetes{
+					Role:           outputs.AuthRole,
+					ServiceAccount: "default",
+					TokenAudiences: []string{"vault"},
+				},
+			},
+		},
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "vaultauth-test-jwt-serviceaccount",
+				Namespace: testK8sNamespace,
+			},
+			Spec: secretsv1alpha1.VaultAuthSpec{
+				Namespace: testVaultNamespace,
+				Method:    "jwt",
+				Mount:     "jwt",
+				Jwt: &secretsv1alpha1.VaultAuthConfigJwt{
+					Role:           outputs.AuthRole,
+					ServiceAccount: "default",
+					TokenAudiences: []string{"vault"},
+				},
+			},
+		},
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "vaultauth-test-jwt-secret",
+				Namespace: testK8sNamespace,
+			},
+			Spec: secretsv1alpha1.VaultAuthSpec{
+				Namespace: testVaultNamespace,
+				Method:    "jwt",
+				Mount:     "jwt",
+				Jwt: &secretsv1alpha1.VaultAuthConfigJwt{
+					Role: outputs.AuthRole,
+					SecretKeyRef: &secretsv1alpha1.SecretKeySelector{
+						Name: secretName,
+						Key:  secretKey,
+					},
+				},
+			},
+		},
+	}
+	expectedData := map[string]interface{}{"foo": "bar"}
+
+	// Apply all of the Auth Methods
+	for _, a := range auths {
+		require.Nil(t, crdClient.Create(ctx, a))
+		created = append(created, a)
+	}
+	secrets := []*secretsv1alpha1.VaultStaticSecret{}
+
+	// create the VSS secrets
+	for _, a := range auths {
+		dest := fmt.Sprintf("kv-%s", a.Name)
+		secretName := fmt.Sprintf("test-secret-%s", a.ObjectMeta.Name)
+		secrets = append(secrets,
+			&secretsv1alpha1.VaultStaticSecret{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      secretName,
+					Namespace: testK8sNamespace,
+				},
+				Spec: secretsv1alpha1.VaultStaticSecretSpec{
+					VaultAuthRef: a.ObjectMeta.Name,
+					Namespace:    testVaultNamespace,
+					Mount:        testKvv2MountPath,
+					Type:         consts.KVSecretTypeV2,
+					Name:         dest,
+					Destination: secretsv1alpha1.Destination{
+						Name:   dest,
+						Create: true,
+					},
+				},
+			})
+	}
+
+	putKV := func(t *testing.T, vssObj *secretsv1alpha1.VaultStaticSecret) {
+		_, err := vClient.KVv2(testKvv2MountPath).Put(ctx, vssObj.Spec.Name, expectedData)
+		require.NoError(t, err)
+	}
+
+	deleteKV := func(t *testing.T, vssObj *secretsv1alpha1.VaultStaticSecret) {
+		require.NoError(t, vClient.KVv2(testKvv2MountPath).Delete(ctx, vssObj.Spec.Name))
+	}
+
+	assertSync := func(t *testing.T, obj *secretsv1alpha1.VaultStaticSecret) {
+		secret, err := waitForSecretData(t, ctx, crdClient, 30, 1*time.Second, obj.Spec.Destination.Name,
+			obj.ObjectMeta.Namespace, expectedData)
+		assert.NoError(t, err)
+		assertSyncableSecret(t, obj,
+			"secrets.hashicorp.com/v1alpha1",
+			"VaultStaticSecret", secret)
+	}
+
+	for x, tt := range auths {
+		t.Run(tt.Spec.Method, func(t *testing.T) {
+			putKV(t, secrets[x])
+			require.Nil(t, crdClient.Create(ctx, secrets[x]))
+			assertSync(t, secrets[x])
+			t.Cleanup(func() {
+				deleteKV(t, secrets[x])
```

Comment:
I don't see where we are cleaning up the `VaultStaticSecrets`, we should probably add that if it is not already being done somewhere else.

Reply:
Thanks for catching that!